server/app/featureflags/FeatureFlags.java

```diff
@@ -85,60 +74,56 @@ public boolean allowCiviformAdminAccessPrograms(Request request) {
   public boolean showCiviformImageTagOnLandingPage(Request request) {
     return getFlagEnabled(request, SHOW_CIVIFORM_IMAGE_TAG_ON_LANDING_PAGE);
   }
 
   // If the UI can show a read only view of a program. Without this flag the
   // only way to view a program is to start editing it.
+  // TODO(#4437): remove and have clients call getFlagEnabled directly.
   public boolean isReadOnlyProgramViewEnabled() {
-    return config.getBoolean(PROGRAM_READ_ONLY_VIEW_ENABLED);
+    return config.getBoolean(PROGRAM_READ_ONLY_VIEW_ENABLED.toString());
   }
 
+  // TODO(#4437): remove and have clients call getFlagEnabled directly.
   public boolean isReadOnlyProgramViewEnabled(Request request) {
     return getFlagEnabled(request, PROGRAM_READ_ONLY_VIEW_ENABLED);
   }
 
+  // TODO(#4437): remove and have clients call getFlagEnabled directly.
   public boolean isEsriAddressCorrectionEnabled(Request request) {
     return getFlagEnabled(request, ESRI_ADDRESS_CORRECTION_ENABLED);
   }
 
+  // TODO(#4437): remove and have clients call getFlagEnabled directly.
   public boolean isEsriAddressServiceAreaValidationEnabled(Request request) {
     return getFlagEnabled(request, ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ENABLED);
   }
 
+  // TODO(#4437): remove and have clients call getFlagEnabled directly.
   public boolean isIntakeFormEnabled(Request request) {
     return getFlagEnabled(request, INTAKE_FORM_ENABLED);
   }
 
+  // TODO(#4437): remove and have clients call getFlagEnabled directly.
   public boolean isNongatedEligibilityEnabled(Request request) {
     return getFlagEnabled(request, NONGATED_ELIGIBILITY_ENABLED);
   }
 
+  // TODO(#4437): remove and have clients call getFlagEnabled directly.
   public boolean isPhoneQuestionTypeEnabled(Request request) {
     return getFlagEnabled(request, PHONE_QUESTION_TYPE_ENABLED);
   }
 
-  public ImmutableMap<String, Boolean> getAllFlags(Request request) {
-    return ImmutableMap.of(
-        ALLOW_CIVIFORM_ADMIN_ACCESS_PROGRAMS,
-        allowCiviformAdminAccessPrograms(request),
-        SHOW_CIVIFORM_IMAGE_TAG_ON_LANDING_PAGE,
-        showCiviformImageTagOnLandingPage(request),
-        PROGRAM_ELIGIBILITY_CONDITIONS_ENABLED,
-        isProgramEligibilityConditionsEnabled(request),
-        PROGRAM_READ_ONLY_VIEW_ENABLED,
-        isReadOnlyProgramViewEnabled(request),
-        ESRI_ADDRESS_CORRECTION_ENABLED,
-        isEsriAddressCorrectionEnabled(request),
-        ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ENABLED,
-        isEsriAddressServiceAreaValidationEnabled(request),
-        INTAKE_FORM_ENABLED,
-        isIntakeFormEnabled(request),
-        NONGATED_ELIGIBILITY_ENABLED,
-        isNongatedEligibilityEnabled(request),
-        PHONE_QUESTION_TYPE_ENABLED,
-        isPhoneQuestionTypeEnabled(request));
+  public Map<FeatureFlag, Boolean> getAllFlagsSorted(Request request) {
+    Map<FeatureFlag, Boolean> map = new TreeMap<>(Comparator.comparing(FeatureFlag::toString));
```

Comment:
Does it matter that TreeMap is not thread-safe/synchronized? Would it be better to pass out an `ImmutableSortedMap`? https://guava.dev/releases/15.0/api/docs/com/google/common/collect/ImmutableSortedMap.html

Reply:
Also, the calling method is depending on this map being sorted, but nothing about the return type enforces that it's sorted.

Reply:
I don't think thread safety matters here, but I'm a fan of ImmutableSortedMap! Changed.